app/web_framework/cluster_base_handler.py

```diff
@@ -192,3 +192,14 @@ def _is_request_origin_allowed(self, request_origin):
         self._logger.debug('Origin "{}" did not match cors_allowed_origins_regex conf value of "{}". '
                            'Not setting Access-Control-Allow-Origin header.', request_origin, allowed_origins_regex)
         return False
+
+
+class ItemNotFoundHandler(ClusterBaseAPIHandler):
+    """
+    ClusterBaseAPIHandler that raises an ItemNotFoundError when it's instantiated.
+    """
+    def initialize(self, route_node=None, **kwargs):
+        super().initialize(route_node)
+
+    def get(self):
+        raise ItemNotFoundError
```

Comment:
Since users will only get this message when they try to make an API call to an endpoint that doesn't support the requested API version, should we give a more detailed error message explaining that's the case? Something like

```
raise ItemNotFoundError('Requested route is not compatible with requested API version.')
```